evap/static/ts/src/csrf-utils.ts

```diff
@@ -28,6 +28,15 @@ $.ajaxSetup({
     },
 });
 
+const inputElement = document.createElement("input");
+inputElement.setAttribute("type", "hidden");
+inputElement.setAttribute("name", "csrfmiddlewaretoken");
+inputElement.setAttribute("value", csrftoken);
+
+for (const form of document.forms) {
+    form.insertAdjacentElement("afterbegin", inputElement.cloneNode() as Element);
+}
```

Comment:
After reading through django's documentation about csrf protection, two more considerations:
- We do not need this if the form method is safe. In staff_semester_view.html, we currently have a form with method="GET". I think this would add the csrftoken as a GET parameter, so the token would be visible on the users screen (in the url bar), making screen sharing unsafe. I'd say we want to prevent that.
- [django documentation on CSRF token](https://docs.djangoproject.com/en/4.0/ref/csrf/#how-to-use-it):
  > This should not be done for POST forms that target external URLs, since that would cause the CSRF token to be leaked, leading to a vulnerability.
  
  If we ever have any form with `action="https://hpi.de/grade-download"` or whatever, we would expose that token. I quickly grepped the source code and currently, I couldn't find any forms with external targets.
  
So, I'd add some kind of check that we're sending a POST request, and it's going to our server. In the existing ajax code, [we check that we have an unsafe method and its not a crossdomain request](https://github.com/e-valuation/EvaP/blob/main/evap/static/ts/src/csrf-utils.ts#L25) before attaching the CSRFToken Header.

Checking the method should be a simple string comparison. Checking for cross domain requests is trickier, and the [number one entry on google, linking to stackoverflow](https://stackoverflow.com/questions/10687099/how-to-test-if-a-url-string-is-absolute-or-relative) makes it look like a futile undertaking, but I feel like abusing the [URL api](https://developer.mozilla.org/en-US/docs/Web/API/URL) should work. This is also what the [5th answer](https://stackoverflow.com/a/57047786/12345551) suggests, after scrolling 7 browser pages.

Reply:
Very good points, thank you for going in-depth on this. I've fixed this in the last commit.

Reply:
...I just learned you can override the form action and method using `formaction` and `formmethod` on the submit buttons. Even though we're not using that, I added support for that. While doing that, I also discovered all this can be done lazily  which is nicer in my opinion.

Reply:
If all this becomes too much complexity, I'm fine with not merging it!